x/connectivity/connectivity.go

```diff
@@ -46,6 +51,23 @@ func (err *ConnectivityError) Unwrap() error {
 	return err.Err
 }
 
+// MarshalJSON implements the json.Marshaler interface.
+func (err *ConnectivityError) MarshalJSON() ([]byte, error) {
```

Comment:
I'd like to keep the connectivity library decoupled from a serialization format. It's a library that anyone can use to report in whatever way they want. Or not reported. For instance, the outline client and cli use the connectivity test, but there's no JSON ever generated.

The serialization should be in the application code.

We do want to make it easier for apps to report in a standard format. Let's first add the logic to an app, see how it should look like in practice, then we can move it to a new package for reuse in another app. Perhaps the new package could be `appconnectivity` or `appreport`?

But before creating a library, it's important to see the code first in a real app like the cli or the gui app.

Reply:
Got it! This was mainly for testing purposing and I didn't intent to include in the ready for review code. I appreciate the feedback. I am doing incremental pushes to this branch and will probably evolve the title and items to include.